helix-term/src/main.rs

```diff
@@ -91,7 +91,16 @@ FLAGS:
     }
 
     let config = match std::fs::read_to_string(conf_dir.join("config.toml")) {
-        Ok(config) => merge_keys(toml::from_str(&config)?),
+        Ok(config) => toml::from_str(&config)
```

Comment:
Right now unknown fields are deserialized into the config, which can allow misspelled config keys to fail silently (eg. `line-numbet = "relative"`). Serde's [`deny_unknown_fields`](https://serde.rs/container-attrs.html#deny_unknown_fields) can come handy here.